test/fuzz/parser_fuzzers.go

```diff
@@ -0,0 +1,337 @@
+func FuzzParseEnv(data []byte) int {
+	f := fuzz.NewConsumer(data)
+
+	// Create a pseudo-random spec.
+	// Will be used as argument to the fuzz target
+
+	// length of slice:
+	qty, err := f.GetInt()
+	if err != nil {
+		return 0
+	}
+	ss := make([]string, qty, qty)
+
+	// fill slice with values
+	for i := 0; i < qty; i++ {
+		s, err := f.GetString()
+		if err != nil {
+			return 0
+		}
+		ss = append(ss, s)
+	}
+	var r io.Reader
+	_, _, _, _ = envutil.ParseEnv(ss, r)
+	return 1
+}
+
+// FuzzParseQOSReserve implements a fuzzer
+// that targets cm.ParseQOSReserved
+func FuzzParseQOSReserve(data []byte) int {
+	f := fuzz.NewConsumer(data)
+
+	// Create a pseudo-random map.
+	// Will be used as argument to the fuzz target
+	m := make(map[string]string)
+	err := f.FuzzMap(&m)
+	if err != nil {
+		return 0
+	}
+	_, _ = cm.ParseQOSReserved(m)
+	return 1
+}
+
+// FuzzParseClientGenTags implements a fuzzer
+// that targets util.ParseClientGenTags
+func FuzzParseClientGenTags(data []byte) int {
+	f := fuzz.NewConsumer(data)
+
+	// Create a pseudo-random lines
+	// Will be used as arguemnt to the fuzz target
+
+	// length of slice:
+	qty, err := f.GetInt()
+	if err != nil {
+		return 0
+	}
+	lines := make([]string, qty, qty)
+
+	// fill slice with values:
+	for i := 0; i < qty; i++ {
+		line, err := f.GetString()
+		if err != nil {
+			return 0
+		}
+		lines = append(lines, line)
+	}
+	_, _ = util.ParseClientGenTags(lines)
+	return 1
+}
+
+// FuzzParseThresholdConfig implements a fuzzer
+// that targets eviction.ParseThresholdConfig
+func FuzzParseThresholdConfig(data []byte) int {
+	f := fuzz.NewConsumer(data)
+
+	// Create a pseudo-random allocatableConfig
+	// Will be used as argument to the fuzz target
+
+	// length of slice:
+	qty, err := f.GetInt()
+	if err != nil {
+		return 0
+	}
+	allocatableConfig := make([]string, qty, qty)
+
+	// fill slice with values:
+	for i := 0; i < qty; i++ {
+		conf, err := f.GetString()
+		if err != nil {
+			return 0
+		}
+		allocatableConfig = append(allocatableConfig, conf)
+	}
+
+	// Create a pseudo-random evictionHard.
+	// Will be used as argument to the fuzz target
+	// evictionHard
+	eh := make(map[string]string)
+	err = f.FuzzMap(&eh)
+	if err != nil {
+		return 0
+	}
+
+	// Create a pseudo-random evictionSoft.
+	// Will be used as argument to the fuzz target
+	// evictionSoft
+	es := make(map[string]string)
+	err = f.FuzzMap(&es)
+	if err != nil {
+		return 0
+	}
+
+	// Create a pseudo-random evictionSoftGracePeriod.
+	// Will be used as argument to the fuzz target
+	// evictionSoftGracePeriod
+	esgp := make(map[string]string)
+	err = f.FuzzMap(&esgp)
+	if err != nil {
+		return 0
+	}
+
+	// Create a pseudo-random evictionMinimumReclaim.
+	// Will be used as argument to the fuzz target
+	// evictionMinimumReclaim
+	emr := make(map[string]string)
+	err = f.FuzzMap(&emr)
+	if err != nil {
+		return 0
+	}
+
+	_, _ = eviction.ParseThresholdConfig(allocatableConfig, eh, es, esgp, emr)
```

Comment:
This is not a good fuzzing target. Input is already basically structured and is trusted.

Reply:
Removed